onnxruntime/core/session/inference_session.cc

```diff
@@ -1619,6 +1623,8 @@ Status InferenceSession::Run(const RunOptions& run_options,
   std::vector<IExecutionProvider*> exec_providers_to_stop;
   exec_providers_to_stop.reserve(execution_providers_.NumProviders());
 
+  std::vector<AllocatorPtr> arenas_to_shrink;
+
```

Comment:
nit: could be a vector of IAllocator (or IArenaAllocator) references to avoid the shared_ptr copy overhead. not sure if anyone would care about the perf cost of that though.

Reply:
I move the shared_ptr (Pranav's comment below) - That should avoid shared_ptr copy overhead. Is that a bit neater than creating a vector of IAllocator ?

Reply:
It's not perf critical code so any of these approaches can be used. Our usage of the AllocatorPtr alias has hidden the underlying std::shared_ptr cost in lots of places anyway.